plugins/org.jboss.tools.openshift.io.ui/src/main/java/org/jboss/tools/openshift/io/internal/ui/DefaultLoginProvider.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.io.internal.ui;
+
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.widgets.Display;
+import org.jboss.tools.openshift.io.core.LoginProvider;
+import org.jboss.tools.openshift.io.core.LoginResponse;
+import org.jboss.tools.openshift.io.core.model.ICluster;
+import org.jboss.tools.openshift.io.core.model.IAccount;
+import org.jboss.tools.openshift.io.internal.ui.dialog.BrowserBasedLoginDialog;
+
+/**
+ * Login provider that will launch a browser to perform the login and extract
+ * the JSON.
+ * 
+ */
+public class DefaultLoginProvider implements LoginProvider {
+
+	@Override
+	public LoginResponse login(ICluster cluster, IAccount account) {
+		if (null == Display.getCurrent()) {
+			final LoginResponse response[] = new LoginResponse[1];
+			Display.getDefault().syncExec(new Runnable() {
+				@Override
+				public void run() {
+					response[0] = loginInUI(cluster, account);
+				}
+			});
+			return response[0];
+			
+		} else {
+			return loginInUI(cluster, account);
+		}
+	}
+
+	public LoginResponse loginInUI(ICluster cluster, IAccount account) {
```

Comment:
Account is never used? This looks like it's handled otherwise in this specific case or it's not used and thus the interface should be changed, too

Reply:
No there is no way to pass account info to the browser. But if we switch to a full HTTP lib implementation, then account would be used.